test_CurrentReadings.cpp

```diff
@@ -0,0 +1,41 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+
+#include "catch.hpp"
+#include "BMS_CurRead.h"
+
+//Test Driven Ranges
+
+TEST_CASE("Case for detecting number of continuous range in the reading array given:Failing") {
+  int ChargingCurrentReadings[] = {3, 3, 5, 4, 10, 11, 12};
+  int numOfCCReadings = sizeof(ChargingCurrentReadings) / sizeof(ChargingCurrentReadings[0]);
+  RangeReading *ArrofRange= DetectContinuousRange(ChargingCurrentReadings, numOfCCReadings);
+  REQUIRE(NumberofContinuousRange(ChargingCurrentReadings, numOfCCReadings) == 3);

```

Comment:
Here the test case has a mistake, hence it is failing. Make a test, where the test code is correct but it fails because the implementation is missing/wrong. E.g., try thinking of error conditions.